codecov_cli/commands/process_test_results.py

```diff
@@ -130,23 +145,38 @@ def process_test_results(
     # GITHUB_REF is documented here: https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
     pr_number = ref.split("/")[2]
 
-    create_github_comment(provider_token, slug, pr_number, message)
-
+    create_github_comment(provider_token, slug, pr_number, message, matrix)
 
-def create_github_comment(token, repo_slug, pr_number, message):
-    url = f"https://api.github.com/repos/{repo_slug}/issues/{pr_number}/comments"
 
+def create_github_comment(token, repo_slug, pr_number, message, matrix):
     headers = {
         "Accept": "application/vnd.github+json",
         "Authorization": f"Bearer {token}",
         "X-GitHub-Api-Version": "2022-11-28",
     }
-    logger.info("Posting github comment")
 
-    log_warnings_and_errors_if_any(
-        send_post_request(url=url, data={"body": message}, headers=headers),
-        "Posting test results comment",
-    )
+    url = f"https://api.github.com/repos/{repo_slug}/issues/{pr_number}/comments"
+    # list comments
+    result = send_get_request(url=url, headers=headers)
+    comments = loads(result.text)
+    if matrix is not None:
+        matrix = matrix.strip()
+    for comment in comments:
+        if f"<!-- Codecov comment for {matrix} -->" in comment:
```

Comment:
Genius. Simple yet effective.

Reply:
[This is not novel haha](https://github.com/actions-cool/maintain-one-comment). But i'm thinking we may want to mention the matrix combination this comment is for, in the comment.